java/ingest/ingest-status-store/src/main/java/sleeper/ingest/status/store/task/IngestTaskStatusStoreFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.ingest.status.store.task;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.ingest.task.IngestTaskStatusStore;
+
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.INGEST_STATUS_STORE_ENABLED;
+
+public class IngestTaskStatusStoreFactory {
+
+    private IngestTaskStatusStoreFactory() {
+    }
+
+    public static IngestTaskStatusStore getStatusStore(AmazonDynamoDB dynamoDB, InstanceProperties properties) {
+        if (Boolean.TRUE.equals(properties.getBoolean(INGEST_STATUS_STORE_ENABLED))) {
```

Comment:
We dont need this boxed boolean comparison anymore, as the SleeperProperties.getBoolean function returns a primitive boolean